.github/workflows/cross-maven.yml

```diff
@@ -0,0 +1,67 @@
+#
+# Copyright Debezium Authors
+#
+# Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Cross CI workflow is triggered on a `pull_request` event.
+# It builds the Debezium core before building the Db2 connector basing on either the `master` branch or the `pull_request` branch.
+# If a `pull_request` with same branch name is found in the core repository, then the core build of this `pull_request` will be based on that.
+# Otherwise the core build of this `pull_request` will be based on the `master` branch of Debezium core repository.
+
+name: Cross Maven CI
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths-ignore:
+      - '*.md'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout db2 repository
+        uses: actions/checkout@v2
+        with:
+          path: db2
+      - name: Check if pull request branch exists in debezium main repository
+        id: branch
+        env:
+          branch_name: ${{ github.head_ref }}
+        run: |
+          curl --silent -X "GET" https://api.github.com/repos/debezium/debezium/pulls | jq '.[] | {branch: .head.ref}' | jq -r '.branch' >> SORTED_PULLS.txt
+
+          while IFS=" " read -r BRANCH;
+          do
+            if grep -q "$branch_name" <<< "$BRANCH"; then
+              echo "::set-output name=BRANCH_FOUND::true"
+            fi
+          done < SORTED_PULLS.txt
+      - name: Checkout core repository with pull request branch
+        if: ${{ steps.branch.outputs.BRANCH_FOUND == 'true' }}
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ github.event.pull_request.user.login }}/debezium
+          ref: ${{ github.head_ref }}
+          path: core
+      - name: Checkout core repository with default branch
```

Comment:
One more question. When a PR is created the user chooses on which branch it should be applied - usually master/main but can be somehing else in case of backporting like 1.6 etc. Could the default branch be not the `master/main` but the one on which the PR is intended to be applied?

Reply:
@jpechane yes, it can be applied.

Reply:
@jpechane One question bugging me, when I created this PR I applied it against the `main` branch of the Db2 repo. But the default branch to build the core can't be `main`, as the debezium upstream default branch is `master`. I am kind of confused here, as you can see the build failed https://github.com/debezium/debezium-connector-db2/runs/3482093060?check_suite_focus=true

Reply:
Yes, that's correct. We renamed `master` versions in non-core repos due to language restrictions but doing that for core would be much more complicated.

Reply:
@jpechane yes, so I think we have to by default provide `master` like it was earlier. This intended branch name is not doable atm.